src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SEGMENT_LENGTH = 100;
+    private static final int FIGURE_QUANTITY = 5;
+    private static final int DEFAULT_NUMBER = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(FIGURE_QUANTITY)) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(),
+                        random.nextDouble() * MAX_SEGMENT_LENGTH);
```

Comment:
you can extract this to a separate method like:
```
private double getRandomSegment() {
   return  random.nextDouble() * MAX_SEGMENT_LENGTH);
}
```

and then use it in all places